pkg/util/spanlogger/spanlogger.go

```diff
@@ -0,0 +1,85 @@
+package spanlogger
```

Comment:
I copied this from Cortex with some changes for passing in a custom `log.Logger`. I don't want to keep this in the Agent forever and I want to contribute it back upstream ASAP.